test/jdk/java/awt/regtesthelpers/PassFailJFrame.java

```diff
@@ -0,0 +1,156 @@
+     * @param instructions    specified instruction that user should follow.
+     * @param maxRowLength    number of visible rows of the JTextArea where the
+     *                        instruction is show.
+     * @param maxStringLength number of columns of the instructional JTextArea
+     * @param timeoutMinutes  timeout of the test where time is specified in
+     *                        minutes.
+     * @throws HeadlessException
+     * @throws InterruptedException
+     * @throws InvocationTargetException
+     */
+    public PassFailJFrame(String title, String instructions,
+                          int maxRowLength, int maxStringLength,
+                          int timeoutMinutes) throws HeadlessException,
+            InterruptedException, InvocationTargetException {
+        super(title);
+        this.maxRowLength = maxRowLength;
+        this.maxStringLength = maxStringLength;
+        this.timeoutMinutes = timeoutMinutes;
+
+        SwingUtilities.invokeAndWait(() -> {
+            setLayout(new BorderLayout());
+            instructionsText = new JTextArea(instructions, maxRowLength, maxStringLength);
+            instructionsText.setEditable(false);
+            instructionsText.setFocusable(false);
+            add(instructionsText, BorderLayout.NORTH);
+
+            JButton btnPass = new JButton("Pass");
+            btnPass.addActionListener((e) -> latch.countDown());
+
+            JButton btnFail = new JButton("Fail");
+            btnFail.addActionListener((e) -> getFailureReason());
+
+            JPanel buttonsPanel = new JPanel();
+            buttonsPanel.add(btnPass);
+            buttonsPanel.add(btnFail);
+
+            add(buttonsPanel, BorderLayout.SOUTH);
+            pack();
+            setLocation(10, 10);
+            setVisible(true);
+        });
+    }
+
+    /**
+     * Wait for the user decision i,e user selects pass or fail button.
+     * If user does not select pass or fail button then the test waits for
+     * the specified timeoutMinutes period and the test gets timeout.
+     * This method should not be invoke with in EDT.
+     *
+     * @throws InterruptedException
+     * @throws InvocationTargetException
+     */
+    public void awaitAndCheck() throws InterruptedException, InvocationTargetException {
+        boolean timeoutHappened = !latch.await(this.timeoutMinutes,
+                TimeUnit.MINUTES);
+
+        SwingUtilities.invokeAndWait(() -> dispose());
+
+        if (timeoutHappened) {
+            throw new RuntimeException("Test timed out!");
+        }
+        if (failed) {
+            throw new RuntimeException("Test failed! : " + testFailedReason);
+        }
+    }
+
+    public void tearDown() throws InterruptedException, InvocationTargetException {
+        SwingUtilities.invokeAndWait(() -> {
+            for (Frame f : JFrame.getFrames()) {
+                System.out.println(f.getTitle() + " is getting disposed");
+                f.dispose();
+            }
```

Comment:
I am still not sure about this. I mean it doesn't include Window or Dialog .. 
I think that until you can prove otherwise it should destroy only the framework, not start being part of the test.

Reply:
We can dispose just the instruction frame and let the user destroy the top levels that is created by the test.